src/main/java/reactor/netty/http/client/HttpClientConnect.java

```diff
@@ -534,7 +535,7 @@ public SocketAddress get() {
 
 		Publisher<Void> requestWithBody(HttpClientOperations ch) {
 			try {
-				ch.resourceUrl = toURI.toExternalForm();
+				ch.resourceUrl = new URI(toURI.toExternalForm());
```

Comment:
Please keep this as String, because `resourceUrl()` might be called many times while the web socket URI is needed just once for the handshake

Reply:
Fixed!